File: backend/tour-design-api/src/main/java/com/tourdesign/platform/service/ClientService.java
package com.tourdesign.platform.service;

import com.tourdesign.platform.model.ClientModel;
import java.util.List;
import java.util.Optional;

public interface ClientService {

    List<ClientModel> list();

    ClientModel create(ClientModel obj);

    Optional<ClientModel> search(Long id);

    List<ClientModel> searchByNameOrLastname(String query);

    Optional<ClientModel> update(Long id, ClientModel obj);

    boolean delete(Long id);
}

File: frontend/tour-design-web/src/app/services/client.service.ts
import { Injectable } from '@angular/core';
import { ApiService } from './api.service';
import { ClientModel } from '../models/client.model';
import { Observable } from 'rxjs';

@Injectable({
  providedIn: 'root'
})
export class ClientService {
  private baseUrl = '/client';

  constructor(private api: ApiService) {}

  list(): Observable<ClientModel[]> {
    return this.api.get<ClientModel[]>(this.baseUrl);
  }

  search(id: number): Observable<ClientModel> {
    return this.api.get<ClientModel>(`${this.baseUrl}/${id}`);
  }

  searchByQuery(query: string): Observable<ClientModel[]> {
    return this.api.get<ClientModel[]>(`${this.baseUrl}/searchByQuery?query=${query}`);
  }

  create(obj: ClientModel): Observable<ClientModel> {
    return this.api.post<ClientModel>(this.baseUrl, obj);
  }

  update(id: number, obj: ClientModel): Observable<ClientModel> {
    return this.api.put<ClientModel>(`${this.baseUrl}/${id}`, obj);
  }

  delete(id: number): Observable<void> {
    return this.api.delete<void>(`${this.baseUrl}/${id}`);
  }
}